orcidhub-core/authcontroller.py

```diff
@@ -393,3 +437,265 @@ def reset_db():
+                    "host": "sandbox.orcid.org",
+                    "uri": "http://sandbox.orcid.org/client/""" + user.organisation.orcid_client_id + """\"
+                }
+            },
+            "path": "",
+            "organization": {
+                "disambiguated-organization": null,
+                "name": \"""" + form.institution.data + """\",
+                "address": {
+                    "city": \"""" + form.city.data + """\",
+                    "region": \"""" + form.state.data + """\",
+                    "country": \"""" + form.country.data + """\"
+                }
+            },
+            "end-date": {
+                "month": {
+                    "value": \"""" + str(form.end_date.data.month).zfill(2) + """\"
+                },
+                "day": {
+                    "value": \"""" + str(form.end_date.data.day).zfill(2) + """\"
+                },
+                "year": {
+                    "value": \"""" + str(form.end_date.data.year) + """\"
+                }
+            },
+            "role-title": \"""" + form.role.data + """\"
+        }"""
+
+        jsondata = json.loads(s)
+        print(jsondata)
+        resp = client.post('https://api.sandbox.orcid.org/v2.0/' + user.orcid + '/employment',
+                           headers=headers, json=jsondata)
+        if (resp.status_code == 200 or resp.status_code == 201):
+            affiliation = User_Organisation_affiliation.create(user=user, organisation=user.organisation,
+                                                               start_date=datetime(int(form.start_date.data.year),
+                                                                                   int(form.start_date.data.month),
+                                                                                   int(form.start_date.data.day)),
+                                                               end_date=datetime(int(form.end_date.data.year),
+                                                                                 int(form.end_date.data.month),
+                                                                                 int(form.end_date.data.day)),
+                                                               department_name=form.department.data,
+                                                               department_city=form.city.data,
+                                                               role_title=form.role.data,
+                                                               put_code=int(
+                                                                   resp.headers['Location'].rsplit('/', 1)[-1]),
+                                                               path=resp.headers['Location'])
+            affiliation.save()
+            flash("""Employment Details has been added successfully!!""", "success")
+        return render_template("addEmploymentDetails.html", form=form)
+
+
+@app.route("/viewempdetails")
```

Comment:
for simplicity, you can utilise **nametuple** as proxy model of ORCID API. See:

https://github.com/Royal-Society-of-New-Zealand/NZ-ORCID-Hub/blob/emp/orcidhub-core/views.py#L46
https://github.com/Royal-Society-of-New-Zealand/NZ-ORCID-Hub/blob/emp/orcidhub-core/views.py#L79
https://github.com/Royal-Society-of-New-Zealand/NZ-ORCID-Hub/blob/emp/orcidhub-core/views.py#L89